smac/smac_cli.py

```diff
@@ -113,29 +122,50 @@ def main_cli(self):
+        tmp_scen = Scenario(tmp_scen, cmd_args={'output_dir':''})
         # Check for folder and files
         rh_path = os.path.join(args_.restore_state, "runhistory.json")
         stats_path = os.path.join(args_.restore_state, "stats.json")
-        traj_path = os.path.join(args_.restore_state, "traj_aclib2.json")
+        traj_path_aclib = os.path.join(args_.restore_state, "traj_aclib2.json")
+        traj_path_old = os.path.join(args_.restore_state, "traj_old.csv")
         scen_path = os.path.join(args_.restore_state, "scenario.txt")
         if not os.path.isdir(args_.restore_state):
            raise FileNotFoundError("Could not find folder from which to restore.")
         # Load runhistory and stats
         rh = RunHistory(aggregate_func=None)
-        rh.load_json(rh_path, scen.cs)
-        root_logger.debug("Restored runhistory from %s", rh_path)
-        stats = Stats(scen)
+        rh.load_json(rh_path, tmp_scen.cs)
+        self.logger.debug("Restored runhistory from %s", rh_path)
+        stats = Stats(tmp_scen)  # Need to inject actual scenario later for output_dir!
         stats.load(stats_path)
-        root_logger.debug("Restored stats from %s", stats_path)
-        trajectory = TrajLogger.read_traj_aclib_format(
-            fn=traj_path, cs=scen.cs)
+        self.logger.debug("Restored stats from %s", stats_path)
+        with open(traj_path_aclib, 'r') as traj_fn:
+            traj_list_aclib = traj_fn.readlines()
+        with open(traj_path_old, 'r') as traj_fn:
+            traj_list_old = traj_fn.readlines()
+        return rh, stats, traj_list_aclib, traj_list_old
+
+    def restore_state_after_scen(self, scen, stats, traj_list_aclib,
+                                 traj_list_old):
+        """Finish processing files for state-restoration. The actual scenario
+        needs to be injected into stats, as well as the trajectory dealt with
+        (it is read in, but needs to be written to new output-folder after
+        scenario is constructed."""
+        stats.scenario = scen  # inject actual scen for output_dir
+        # write trajectory-list
+        traj_path_aclib = os.path.join(scen.output_dir, "traj_aclib2.json")
+        traj_path_old = os.path.join(scen.output_dir, "traj_old.csv")
+        with open(traj_path_aclib, 'w') as traj_fn:
+            traj_fn.writelines(traj_list_aclib)
+        with open(traj_path_old, 'w') as traj_fn:
+            traj_fn.writelines(traj_list_old)
+        # read trajectory to retrieve incumbent
+        trajectory = TrajLogger.read_traj_aclib_format(fn=traj_path_aclib, cs=scen.cs)
         incumbent = trajectory[-1]["incumbent"]
-        root_logger.debug("Restored incumbent %s from %s", incumbent, traj_path)
-        # Copy traj if output_dir of specified scenario-file is different than
-        # the output_dir of the scenario-file in the folder from which to restore.
-        if scen.output_dir != InputReader().read_scenario_file(scen_path)['output_dir']:
```

Comment:
Okay, so this behavior is removed (because it probably was unnecessary)?

Reply:
because of the refactoring of the smac-cli-dependencies. now the traj is read in `restore_state_before_scen` and written in `restore_state_after_scen`. this is to make sure, that restoring a run in the same outputfolder does not cause any problems (because the outputfolders are created upon scenario-construction and old folders are moved to folder.OLD, so the paths are not valid anymore. this way, restoring a state is robust against any file- or folder-relocation).